internal/controller/apiextensions/composition/reconciler.go

```diff
@@ -156,20 +158,24 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	}
 
 	var latestRev, existingRev int64
+
+	if lr := composite.LatestRevision(rl.Items); lr != nil {
+		latestRev = composite.LatestRevision(rl.Items).Spec.Revision
+	}
+
 	for i := range rl.Items {
 		rev := &rl.Items[i]
-		if !metav1.IsControlledBy(rev, comp) {
-			continue
-		}
```

Comment:
We're listing composition revisions with the composition name label. Therefore, we don't need to check for the owner.

Reply:
I'd like to continue checking that the Composition is the controller reference of the CompositionRevision. This would catch, for example, the case where:

* Someone made a Composition named `foo`
* Someone deleted said Composition, but somehow the CompositionRevision was not deleted (e.g. a race)
* Someone else made a new, different Composition named `foo`

Reply:
Thanks, @negz for the explanation! I re-add the checks.